src/test/java/core/basesyntax/MyLinkedListTest.java

```diff
@@ -0,0 +1,215 @@
+        exception.expect(IndexOutOfBoundsException.class);
+        myLinkedList.get(3);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void testGetByNegativeIndex() {
+        myLinkedList.get(-1);
+    }
+
+    @Test
+    public void testSetByIndex() {
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        myLinkedList.set(NEW_ITEM, 0);
+        myLinkedList.set(ANOTHER_NEW_ITEM, 1);
+        String actualFirst = myLinkedList.get(0);
+        String actualSecond = myLinkedList.get(1);
+        Assert.assertEquals(NEW_ITEM, actualFirst);
+        Assert.assertEquals(ANOTHER_NEW_ITEM, actualSecond);
+        exception.expect(IndexOutOfBoundsException.class);
+        myLinkedList.set(NEW_ITEM, 3);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void testSetByZeroIndex() {
+        myLinkedList.set(NEW_ITEM, 0);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void testSetByNegativeIndex() {
+        myLinkedList.set(NEW_ITEM, -1);
+    }
+
+    @Test
+    public void testRemoveByIndex() {
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        String actualFirst = myLinkedList.remove(0);
+        String actualSecond = myLinkedList.remove(3);
+        String actualThird = myLinkedList.remove(3);
+        Assert.assertEquals(FIRST_ITEM, actualFirst);
+        Assert.assertEquals(SECOND_ITEM, actualSecond);
+        Assert.assertEquals(THIRD_ITEM, actualThird);
+        exception.expect(IndexOutOfBoundsException.class);
+        myLinkedList.remove(3);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void testRemoveByNegativeIndex() {
+        myLinkedList.remove(-1);
+    }
+
+    @Test
+    public void testRemoveByItem() {
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        String actualFirst = myLinkedList.remove(FIRST_ITEM);
+        String actualSecond = myLinkedList.remove(SECOND_ITEM);
+        String actualThird = myLinkedList.remove(THIRD_ITEM);
+        Assert.assertEquals(FIRST_ITEM, actualFirst);
+        Assert.assertEquals(SECOND_ITEM, actualSecond);
+        Assert.assertEquals(THIRD_ITEM, actualThird);
+        String actualFourth = myLinkedList.remove(NEW_ITEM);
+        Assert.assertNull(actualFourth);
+    }
+
+    @Test
+    public void testSize() {
+        int actualInitialSize = myLinkedList.size();
+        myLinkedList.add(FIRST_ITEM);
+        myLinkedList.add(SECOND_ITEM);
+        myLinkedList.add(THIRD_ITEM);
+        int actualSize = myLinkedList.size();
```

Comment:
you don't need this variable. Use assert immediately
```suggestion
        Assert.assertEquals(3, myLinkedList.size());
```